fe/fe-core/src/main/java/org/apache/doris/external/elasticsearch/EsUtil.java

```diff
@@ -215,20 +184,29 @@ public static List<Column> genColumnsFromEs(EsRestClient client, String indexNam
             column.setUniqueId(-1);
             columns.add(column);
         }
-        for (String key : keys) {
-            JSONObject field = (JSONObject) mappingProps.get(key);
-            Type type;
-            // Complex types are treating as String types for now.
-            if (field.containsKey("type")) {
-                type = toDorisType(field.get("type").toString());
-            } else {
-                type = Type.STRING;
-            }
+        Iterator<String> iterator = mappingProps.fieldNames();
+        while (iterator.hasNext()) {
+            String key = iterator.next();
             Column column = new Column();
             column.setName(key);
             column.setIsKey(true);
             column.setIsAllowNull(true);
             column.setUniqueId(-1);
+            ObjectNode field = (ObjectNode) mappingProps.get(key);
+            Type type;
+            // Complex types are treating as String types for now.
+            if (field.has("type")) {
+                String typeStr = field.get("type").asText();
+                if ("date".equals(typeStr)) {
+                    type = parseEsDateType(column, field);
+                } else {
+                    type = toDorisType(typeStr);
```

Comment:
fill all these codes into the function `toDorisType`? I think toDorisType is only used here